modules/charon-core/src/test/java/org/wso2/charon3/core/protocol/endpoints/GroupResourceManagerTest.java

```diff
@@ -0,0 +1,944 @@
+    private UserManager userManager;
+
+    @BeforeMethod
+    public void setUp() {
+
+        groupResourceManager = new GroupResourceManager();
+        userManager = mock(UserManager.class);
+    }
+
+    public void tearDown() {
+
+    }
+
+    private SCIMResponse getEncodeSCIMExceptionObject(AbstractCharonException exception) {
+
+        JSONEncoder encoder = new JSONEncoder();
+        Map<String, String> responseHeaders = new HashMap<>();
+        responseHeaders.put(SCIMConstants.CONTENT_TYPE_HEADER, SCIMConstants.APPLICATION_JSON);
+        return new SCIMResponse(exception.getStatus(), encoder.encodeSCIMException(exception), responseHeaders);
+    }
+
+    private Group getNewGroup() throws BadRequestException, CharonException, InternalErrorException {
+
+        SCIMResourceTypeSchema schema = SCIMResourceSchemaManager.getInstance().getGroupResourceSchema();
+        JSONDecoder decoder = new JSONDecoder();
+
+        return decoder.decodeResource(NEW_GROUP_SCIM_OBJECT_STRING, schema, new Group());
+    }
+
+    @DataProvider(name = "dataForGetGroupSuccess")
+    public Object[][] dataToGetGroupSuccess() throws CharonException, InternalErrorException, BadRequestException {
+
+        Group group = getNewGroup();
+        String id = group.getId();
+        String name = group.getDisplayName();
+
+        return new Object[][]{
+                {id, null, null, group},
+                {id, null, "members", group},
+                {id, name, null, group},
+        };
+    }
+
+    @Test(dataProvider = "dataForGetGroupSuccess")
+    public void testGetGroupSuccess(String id, String attributes, String excludeAttributes, Object objectGroup)
+            throws BadRequestException, NotFoundException, CharonException, NotImplementedException {
+
+        Group group = (Group) objectGroup;
+
+        SCIMResourceTypeSchema schema = SCIMResourceSchemaManager.getInstance().getGroupResourceSchema();
+
+        Map<String, Boolean> requiredAttributes = ResourceManagerUtil.getOnlyRequiredAttributesURIs(
+                (SCIMResourceTypeSchema) CopyUtil.deepCopy(schema), attributes, excludeAttributes);
+
+        mockStatic(AbstractResourceManager.class);
+
+        when(AbstractResourceManager.getResourceEndpointURL(SCIMConstants.USER_ENDPOINT))
+                .thenReturn(SCIM2_GROUP_ENDPOINT);
+        when(AbstractResourceManager.getEncoder()).thenReturn(new JSONEncoder());
+        when(AbstractResourceManager.getDecoder()).thenReturn(new JSONDecoder());
+        when(userManager.getGroup(id, requiredAttributes)).thenReturn(group);
+
+        SCIMResponse outputScimResponse = groupResourceManager.get(id, userManager, attributes, excludeAttributes);
+        JSONObject obj = new JSONObject(outputScimResponse.getResponseMessage());
+
+        Assert.assertEquals(outputScimResponse.getResponseStatus(), ResponseCodeConstants.CODE_OK);
+        if (attributes == null & excludeAttributes == null) {
```

Comment:
conditions used seems to be not the best option. You can check if the requested attributes are in the object and excluded attributes does not included in the response.

Reply:
Fixed in commit 71483cb81fc57041180aaa8bc31aa99110211062

Reply:
This code block is still there.

Reply:
Removed in : commit 6220cf93a94ca41ba7805932a36440de7362224d